packages/php/email-editor/src/Engine/class-email-editor.php

```diff
@@ -288,7 +288,8 @@ public function load_js_vars() {
 
 		?>
 		<script type="text/javascript"> <?php // phpcs:ignore ?>
-			window.mailpoet_email_editor_current_post_type = <?php echo wp_json_encode( $email_editor_current_post_type ); ?>;
+			window.mailpoet_email_editor_current_post_type = '<?php echo esc_js( $email_editor_current_post_type ); ?>';
+			window.mailpoet_email_editor_current_post_id = <?php echo esc_js( $post->ID ); ?>;
```

Comment:
I'm experiencing a weird issue related to this change. I already have a bunch of emails in the DB. When I create a new email in admin UI the editor loads an older email instead of the new one.
I noticed that the postId in the URL differs from the one we passed to the editor.
![Screenshot 2025-01-29 at 14 39 58](https://github.com/user-attachments/assets/458f6aec-647a-45b9-94c1-8f1bc584ae41)

Is the global $post variable reliable to determine the currently edited post? Perhaps we could move that to the integrator code as I suggested for the other variable where we know the id.

Reply:
Done here https://github.com/mailpoet/mailpoet/pull/6077/commits/12344da42842a10c6e08c007e0e1e414d8d4241d